### ISSUE TITLE
abrt-dump-oops does not respect -x (world readable) option

### ISSUE THREAD

**Issue by hartsjc** [abrt/abrt#1515]
Issue originaly seen in RHEL 7 even with with `PrivateReports = off`; however, present in Fedora too.

Setup for Fedora
```
# rpm -qf /usr/bin/abrt-dump-oops 
abrt-addon-kerneloops-2.14.2-2.fc32.x86_64

# abrt-dump-oops -h 2>&1 | grep -e -x
    -x                          Make the problem directory world readable

# umask
0022
```

And here can see the test directory does NOT have world read.
```
# rm -rf test ; abrt-dump-oops -v -xd test vmcore-dmesg.txt
abrt-dump-oops: Found oopses: 1
abrt-dump-oops: Creating problem directories
abrt-dump-oops: Saving 1 oopses as problem dirs
abrt-dump-oops: Kernel is tainted 'POE'

# ll -d test
drwxr-x---. 3 root root 45 Aug  5 11:53 test
```

And with strace can see explictily calls mkdir with 0750.
```
# rm -rf test ; strace -vfe mkdir abrt-dump-oops -xd test vmcore-dmesg.txt > /dev/null
abrt-dump-oops: Found oopses: 1
abrt-dump-oops: Creating problem directories
mkdir("test/oops-2020-08-05-11:56:44-1770-0", 0750) = -1 ENOENT (No such file or directory)
mkdir("test", 0750)                     = 0
mkdir("test/oops-2020-08-05-11:56:44-1770-0", 0750) = 0
+++ exited with 0 +++
```

**Comment by hartsjc** [comment]
https://github.com/abrt/abrt/blob/master/src/plugins/oops-utils.c#L111
```
111         struct dump_dir *dd = dd_create(path, /*fs owner*/0, DEFAULT_DUMP_DIR_MODE);
```

Changed to:
```
111         struct dump_dir *dd = dd_create(path, /*fs owner*/0, (ABRT_OOPS_WORLD_READABLE ? 0755 : DEFAULT_DUMP_DIR_MODE));
```

Then think we have these excepted perms on direcotry
```
# strace -v -e mkdir abrt-dump-oops -xd test vmcore-dmesg.txt ; ll -d test

mkdir("test", 0755)                     = 0
drwxr-xr-x. 3 root root 47 Aug  6 13:03 test
```

**Comment by hartsjc** [comment]
https://github.com/abrt/abrt/blob/master/src/plugins/oops-utils.c#L111
```
111         struct dump_dir *dd = dd_create(path, /*fs owner*/0, DEFAULT_DUMP_DIR_MODE);
```

Changed to:
```
111         struct dump_dir *dd = dd_create(path, /*fs owner*/0, ((flags & ABRT_OOPS_WORLD_READABLE) ? 0755 : DEFAULT_DUMP_DIR_MODE));
```

Then with `-x` we have rx for world as expected
```
# strace -v -e mkdir abrt-dump-oops -xd test vmcore-dmesg.txt ; ll -d test
mkdir("test/oops-2020-08-06-13:17:03-713244-0", 0755) = -1 ENOENT (No such file or directory)
mkdir("test", 0755)                     = 0
mkdir("test/oops-2020-08-06-13:17:03-713244-0", 0755) = 0
+++ exited with 0 +++
drwxr-xr-x. 3 root root 47 Aug  6 13:17 test
```

And without the `-x` still have the 0750 (no world access)
```
# strace -v -e mkdir abrt-dump-oops -d test vmcore-dmesg.txt ; ll -d test
mkdir("test/oops-2020-08-06-13:16:57-713219-0", 0750) = -1 ENOENT (No such file or directory)
mkdir("test", 0750)                     = 0
mkdir("test/oops-2020-08-06-13:16:57-713219-0", 0750) = 0
+++ exited with 0 +++
drwxr-x---. 3 root root 47 Aug  6 13:16 test
```

**Comment by mgrabovsky** [comment]
Thanks for reporting the problem. I have proposed a fix in #1517.
